### DISCUSSION TITLE
Front End Developer

### DISCUSSION THREAD

**Discussion by KarinaMacedo13** [community/community#18469] [Other Feature Feedback, Questions, & Ideas]
What every front end developer should know?

**Comment by doguedogue** [comment]
I think 🚀
1. git / github
2. html
3. css
4. js
5. frameworks (react, vue, angular)

**Comment by melvincwng** [comment]
@KarinaMacedo13, first of all, I would like to welcome you to Github Community :)

Specifically to answer your question, as a front-end developer, you should learn the following:
1. HTML5, CSS3, JS
2. Frameworks (pick one - either React/Angular/Vue)
3. Cloud tools (e.g. Netlify/Heroku/Vercel/AWS)
4. Postman (API testing)
5. Additional tools - like VSCode or any code editor of your choice.

But don't just stop at front-end! I recommend you to explore some back-end technologies as well, so that you can understand the whole spectrum of full-stack application development!

All the best in your journey as a software developer :)

Hope this answer helps you!

**Comment by lesongvi** [comment]
In my opinion,
- HTML5/CSS3/JavaScript
- Frameworks such as Angular, ReactJS, VueJS
- Responsive Design
- Version Control/Git
- Testing/Debugging
- Browser Developer Tools
- Web Performance
- CSS Preprocessing
- Command Line

**Comment by baysideengineer** [comment]
Hello @KarinaMacedo13, this [roadmap](https://roadmap.sh/frontend) here is a very detailed overview of things you should know or heard of as frontend developer. You can follow the path or create your own based on it. There are also more roadmaps available at [https://roadmap.sh](https://roadmap.sh). Hope this gives you a good overview.

**Comment by brz95** [comment]
HTML5/CSS3
JavaScript
DOM
Webpack
REST API
Frameworks (React, Vue or Angular)
SOLID
HTTP
WebAPI

**Comment by Allencs16** [comment]
I think, a good knowledge in UX, UI, a good base with HTML, CSS and JS if u are web.

**Comment by Minseok0917** [comment]
Trend 
- [state of css](https://stateofcss.com/en-us/)
- [state of js](https://stateofjs.com/en-us/)

**Comment by Ngquang** [comment]
> What every front end developer should know?

I found this a few years ago, but it's still relevant today.

https://roadmap.sh/frontend

With me, it's not just about the technical requirements; you also need to know about color, layout, texture, and other aspects of art.

**Comment by basitmir7** [comment]
Every frontend development should know:
1. HTML and CSS ,should be able to create any design into code.
2. Should know  how to make responsive websites.
3. Should have a good knowledge of UI UX design.
4. Good working knowledge of JavaScript.
5. API integration.
6. testing and debugging.
7. Knowledge of Git
8. According to me frameworks come at last. Being great at core technologies i.e. HTML,CSS and JavaScript, Frameworks will not be a problem to learn but its good to have knowledge of at least one of the popular frontend frameworks like React, Angular and Vue.
